diff --git a/src/Numeric/Interpolator/NodeBilinear.H b/src/Numeric/Interpolator/NodeBilinear.H
index 21a8253fd..b9524bf1b 100644
--- a/src/Numeric/Interpolator/NodeBilinear.H
+++ b/src/Numeric/Interpolator/NodeBilinear.H
@@ -2,7 +2,7 @@
 // Provide an interpolator function that can work with templated fields.
 //
 // :bdg-danger-line:`Warning: This is a low-level class used to add templating functionality to unerlying AMReX classes. Edit at your own risk!`
-//    
+//
 //
 
 
@@ -13,6 +13,7 @@
 #include <AMReX_BCRec.H>
 #include <AMReX_REAL.H>
 #include <AMReX_GpuControl.H>
+#include <AMReX_GpuElixir.H>
 #include <AMReX_Interp_C.H>
 #include <AMReX_Interpolater.H>
 
@@ -39,18 +40,20 @@ public:
         int /*actual_state*/,
         amrex::RunOn runon)
     {
-        // bool run_on_gpu = (runon == amrex::RunOn::Gpu && amrex::Gpu::inLaunchRegion());
+        bool run_on_gpu = (runon == amrex::RunOn::Device && amrex::Gpu::inLaunchRegion());
 
         int num_slope = ncomp * (AMREX_D_TERM(2, *2, *2) - 1);
         const amrex::Box cslope_bx = amrex::enclosedCells(CoarseBox(fine_region, ratio));
         amrex::BaseFab<T> slopefab(cslope_bx, num_slope);
-        // amrex::Elixir slopeeli;
-        // if (run_on_gpu) slopeeli = slopefab.elixir();
+        amrex::Elixir slopeeli;
+        if (run_on_gpu) slopeeli = slopefab.elixir();
 
         amrex::Array4<T const> const& crsearr = crse.const_array();
         amrex::Array4<T> const& finearr = fine.array();
         amrex::Array4<T> const& slopearr = slopefab.array();
 
+        using namespace amrex; // normally avoid this! but must use for now.
+
         AMREX_LAUNCH_HOST_DEVICE_LAMBDA_FLAG(runon, cslope_bx, tbx,
             {
                 amrex::nodebilin_slopes<T>(tbx, slopearr, crsearr, crse_comp, ncomp, ratio);
