diff --git a/src/Model/Solid/InClassOperators.H b/src/Model/Solid/InClassOperators.H
index aeac3bd47..797bd7bd7 100644
--- a/src/Model/Solid/InClassOperators.H
+++ b/src/Model/Solid/InClassOperators.H
@@ -1,4 +1,5 @@
 AMREX_FORCE_INLINE
+AMREX_GPU_HOST_DEVICE
 void operator += (const OP_CLASS &rhs)
 {
 #define X(name) \
@@ -8,6 +9,7 @@ void operator += (const OP_CLASS &rhs)
 }    
 
 AMREX_FORCE_INLINE
+AMREX_GPU_HOST_DEVICE
 OP_CLASS operator * (const Set::Scalar alpha) const
 {
         OP_CLASS ret;
@@ -17,7 +19,7 @@ OP_CLASS operator * (const Set::Scalar alpha) const
 #undef X
     return ret;
 }    
-friend OP_CLASS operator * (const Set::Scalar alpha, const OP_CLASS b);
-friend OP_CLASS operator + (const OP_CLASS a, const OP_CLASS b);
-friend OP_CLASS operator - (const OP_CLASS a, const OP_CLASS b);
-friend bool operator == (const OP_CLASS a, const OP_CLASS b);
+friend AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE OP_CLASS operator * (const Set::Scalar alpha, const OP_CLASS b);
+friend AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE OP_CLASS operator + (const OP_CLASS a, const OP_CLASS b);
+friend AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE OP_CLASS operator - (const OP_CLASS a, const OP_CLASS b);
+friend AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE bool operator == (const OP_CLASS a, const OP_CLASS b);
