diff --git a/src/Integrator/Hydro.cpp b/src/Integrator/Hydro.cpp
index 875462de8..9ec8a85f0 100644
--- a/src/Integrator/Hydro.cpp
+++ b/src/Integrator/Hydro.cpp
@@ -188,9 +188,6 @@ Hydro::Parse(Hydro& value, IO::ParmParse& pp)
     // Gas model (Thermo, Transport, and EOS)
     pp.queryclass<Model::Gas::Gas>("gas", value.gas);
     value.nspecies = value.gas.nspecies;
-    std::cout << value.gas.thermo->model_name() << "\n";
-    std::cout << value.gas.transport->model_name() << "\n";
-    std::cout << value.gas.eos->model_name() << "\n";
     std::cout << value.nspecies << "\n";
 
     std::string prescribedflowmode_str;
