diff --git a/src/Integrator/Integrator.H b/src/Integrator/Integrator.H
index 84ba2536c..73af14dad 100644
--- a/src/Integrator/Integrator.H
+++ b/src/Integrator/Integrator.H
@@ -256,7 +256,8 @@ protected:
 
     // Dynamic timestep adjustment
 
-    struct {
+public:
+    struct dynamictimestep_st{
         // user params
         bool on = false;
         int verbose = -1;
@@ -267,7 +268,7 @@ protected:
         // internal variables
         std::vector<Set::Scalar> dt_limit_min;
         std::vector<Set::Scalar> previous_timesteps;
-    } dynamictimestep; /// Params for the dynamic timestp
+    }; dynamictimestep_st dynamictimestep; /// Params for the dynamic timestp
     void DynamicTimestep_SyncTimeStep(int lev, Set::Scalar dt_min)
     {
         if (!dynamictimestep.on) return;
@@ -465,6 +466,11 @@ protected:
         std::vector<amrex::Box> box;
     } explicitmesh;
 
+#ifdef AMREX_USE_HDF5
+protected:
+    std::string compression_level = "";
+#endif
+
 protected:
     int plot_int = -1;               ///< How frequently to dump plot file (default: never)
     Set::Scalar plot_dt = -1.0;
