diff --git a/src/Model/Propellant/PowerLaw.H b/src/Model/Propellant/PowerLaw.H
index 372668600..15051c837 100644
--- a/src/Model/Propellant/PowerLaw.H
+++ b/src/Model/Propellant/PowerLaw.H
@@ -31,6 +31,14 @@ public:
     }
 
 
+    template<int I = 0>
+    AMREX_GPU_HOST
+    Set::Scalar get (std::string /*property*/)
+    {
+        Util::Abort(INFO);
+        return NAN;
+    }
+
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
     void set_pressure(const Set::Scalar P) 
     {
@@ -45,17 +53,16 @@ public:
     }
 
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_K (const Set::Scalar /*phi*/) {return NAN;}
+    Set::Scalar get_K (const Set::Scalar /*phi*/) const {return NAN;}
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_rho (const Set::Scalar /*phi*/) {return NAN;}
+    Set::Scalar get_rho (const Set::Scalar /*phi*/) const {return NAN;}
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_cp (const Set::Scalar /*phi*/) {return NAN;}
+    Set::Scalar get_cp (const Set::Scalar /*phi*/) const {return NAN;}
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_qdot (const Set::Scalar /*mdot*/, const Set::Scalar /*phi*/) {return NAN;}
+    Set::Scalar get_qdot (const Set::Scalar /*mdot*/, const Set::Scalar /*phi*/) const {return NAN;}
 
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_L (Set::Scalar phi, Set::Scalar /*T*/) 
-    {
+    Set::Scalar get_L (Set::Scalar phi, Set::Scalar /*T*/) const    {
         Set::Scalar L = NAN;
         // if (homogeneousSystem == 1) {
         //     Set::Vector grad_eta = Numeric::Gradient(eta, i, j, k, 0, DX);
