diff --git a/src/Model/Propellant/Propellant.H b/src/Model/Propellant/Propellant.H
index 502dffb08..c00581312 100644
--- a/src/Model/Propellant/Propellant.H
+++ b/src/Model/Propellant/Propellant.H
@@ -35,6 +35,25 @@ private:
 
 public:
 
+    std::string get_name()
+    {
+        return names[selected];
+    }
+
+    template<int I = 0>
+    AMREX_GPU_HOST
+    Set::Scalar get (std::string property)
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I) return std::get<I>(obj).get(property);
+            else return get<I+1>(property);
+        }
+        else Util::Abort(INFO,selected);
+        return NAN;
+    }
+
+
     template<int I = 0>
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
     void set_pressure (Set::Scalar P)
@@ -49,8 +68,7 @@ public:
 
     template<int I = 0>
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_K (const Set::Scalar phi)
-    {
+    Set::Scalar get_K (const Set::Scalar phi) const    {
         if constexpr (I < sizeof...(OBJ))
         {
             if (selected == I) return std::get<I>(obj).get_K(phi);
@@ -62,8 +80,7 @@ public:
 
     template<int I = 0>
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_rho (const Set::Scalar phi)
-    {
+    Set::Scalar get_rho (const Set::Scalar phi) const    {
         if constexpr (I < sizeof...(OBJ))
         {
             if (selected == I) return std::get<I>(obj).get_rho(phi);
@@ -75,8 +92,7 @@ public:
 
     template<int I = 0>
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_cp (const Set::Scalar phi)
-    {
+    Set::Scalar get_cp (const Set::Scalar phi) const    {
         if constexpr (I < sizeof...(OBJ))
         {
             if (selected == I) return std::get<I>(obj).get_cp(phi);
@@ -88,8 +104,7 @@ public:
 
     template<int I = 0>
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_qdot (const Set::Scalar mdot, const Set::Scalar phi) 
-    {
+    Set::Scalar get_qdot (const Set::Scalar mdot, const Set::Scalar phi) const    {
         if constexpr (I < sizeof...(OBJ))
         {
             if (selected == I) return std::get<I>(obj).get_qdot(mdot,phi);
@@ -101,8 +116,7 @@ public:
 
     template<int I = 0>
     AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
-    Set::Scalar get_L (const Set::Scalar phi, const Set::Scalar T) 
-    {
+    Set::Scalar get_L (const Set::Scalar phi, const Set::Scalar T) const    {
         if constexpr (I < sizeof...(OBJ))
         {
             if (selected == I) return std::get<I>(obj).get_L(phi,T);
