diff --git a/src/Set/Matrix4_MajorMinor.H b/src/Set/Matrix4_MajorMinor.H
index 2946fdd25..3dc0d4d4f 100644
--- a/src/Set/Matrix4_MajorMinor.H
+++ b/src/Set/Matrix4_MajorMinor.H
@@ -25,6 +25,7 @@ public:
     //    for (int i = 0; i < 6; i++) data[i] =  in.data[i];
     //}
     AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
     const Scalar & operator () (const int i, const int j, const int k, const int l) const
     {
         int uid = i + 2*j + 4*k + 8*l;
@@ -35,9 +36,10 @@ public:
         else if (uid==14 || uid==13 || uid==11 || uid==7 ) return data[4]; // [0111] [1011] [1101] [1110]
         else if (uid==15 )                                 return data[5]; // [1111]
         else Util::Abort(INFO,"Index out of range");
-        return Set::Garbage;
+        return Set::Garbage();
     }
     AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
     Scalar & operator () (const int i, const int j, const int k, const int l)
     {
         int uid = i + 2*j + 4*k + 8*l;
@@ -48,7 +50,7 @@ public:
         else if (uid==14 || uid==13 || uid==11 || uid==7 ) return data[4]; // [0111] [1011] [1101] [1110]
         else if (uid==15 )                                 return data[5]; // [1111]
         else Util::Abort(INFO,"Index out of range");
-        return Set::Garbage; 
+        return Set::Garbage(); 
     }
     void Print (std::ostream& os)
     {
@@ -116,12 +118,14 @@ public:
         ret.Randomize();
         return ret;
     }
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<2,Sym::MajorMinor> Zero()
     {
         Matrix4<2,Sym::MajorMinor> ret;
         for (int i = 0 ; i < 6; i++) ret.data[i] = 0.0;
         return ret;
     }
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<2,Sym::MajorMinor> Cubic(Set::Scalar C11, Set::Scalar C12, Set::Scalar C44, 
                                             Eigen::Matrix3d R = Eigen::Matrix3d::Identity())
     {
@@ -152,6 +156,7 @@ public:
                     }                
         return ret;
     }
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<2,Sym::MajorMinor> Cubic(Set::Scalar C11, Set::Scalar C12, Set::Scalar C44, 
                                             Set::Scalar phi1, Set::Scalar Phi, Set::Scalar phi2)
     {
@@ -161,6 +166,8 @@ public:
             Eigen::AngleAxisd(phi1, Eigen::Vector3d::UnitX());
         return Matrix4<2,Sym::MajorMinor>::Cubic(C11,C12,C44,R);
     }
+
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<2,Sym::MajorMinor> Cubic(Set::Scalar C11, Set::Scalar C12, Set::Scalar C44, 
                                             Set::Quaternion q)
     {
@@ -233,14 +240,14 @@ public:
     }
 
 
-    friend Eigen::Matrix<Set::Scalar,2,2> operator * (Matrix4<2,Sym::MajorMinor> a, Eigen::Matrix<Set::Scalar,2,2> b);
-    friend bool operator == (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b);
-    friend Matrix4<2,Sym::MajorMinor> operator + (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b);
-    friend Matrix4<2,Sym::MajorMinor> operator - (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b);
-    friend Matrix4<2,Sym::MajorMinor> operator * (Matrix4<2,Sym::MajorMinor> a, Set::Scalar b);
-    friend Matrix4<2,Sym::MajorMinor> operator * (Set::Scalar b, Matrix4<2,Sym::MajorMinor> a);
-    friend Matrix4<2,Sym::MajorMinor> operator / (Matrix4<2,Sym::MajorMinor> a, Set::Scalar b);
-    friend Set::Vector operator * (Matrix4<2,Sym::MajorMinor> a, Set::Matrix3 b);
+    friend AMREX_GPU_HOST_DEVICE Eigen::Matrix<Set::Scalar,2,2> operator * (Matrix4<2,Sym::MajorMinor> a, Eigen::Matrix<Set::Scalar,2,2> b);
+    friend AMREX_GPU_HOST_DEVICE bool operator == (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<2,Sym::MajorMinor> operator + (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<2,Sym::MajorMinor> operator - (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<2,Sym::MajorMinor> operator * (Matrix4<2,Sym::MajorMinor> a, Set::Scalar b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<2,Sym::MajorMinor> operator * (Set::Scalar b, Matrix4<2,Sym::MajorMinor> a);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<2,Sym::MajorMinor> operator / (Matrix4<2,Sym::MajorMinor> a, Set::Scalar b);
+    friend AMREX_GPU_HOST_DEVICE Set::Vector operator * (Matrix4<2,Sym::MajorMinor> a, Set::Matrix3 b);
 };
 AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE 
 bool operator == (Matrix4<2,Sym::MajorMinor> a, Matrix4<2,Sym::MajorMinor> b)
@@ -325,6 +332,7 @@ class Matrix4<3,Sym::MajorMinor>
 public:
     AMREX_GPU_HOST_DEVICE Matrix4() {};
     AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
     const Scalar & operator () (const int i, const int j, const int k, const int l) const
     {
         int uid = i + 3*j + 9*k + 27*l;
@@ -371,9 +379,10 @@ public:
         // [2222]
         else if (uid==80 ) return data[20];
         else Util::Abort(INFO,"Index out of range");
-        return Set::Garbage; 
+        return Set::Garbage(); 
     }
     AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
     Scalar & operator () (const int i, const int j, const int k, const int l)
     {
         int uid = i + 3*j + 9*k + 27*l;
@@ -420,7 +429,7 @@ public:
         // [2222]
         else if (uid==80 ) return data[20];
         else Util::Abort(INFO,"Index out of range");
-        return Set::Garbage; 
+        return Set::Garbage(); 
     }
     void Print (std::ostream& os)
     {
@@ -509,12 +518,15 @@ public:
         ret.Randomize();
         return ret;
     }
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<3,Sym::MajorMinor> Zero()
     {
         Matrix4<3,Sym::MajorMinor> ret;
         for (int i = 0 ; i < 21; i++) ret.data[i] = 0.0;
         return ret;
     }
+
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<3,Sym::MajorMinor> Cubic(Set::Scalar C11, Set::Scalar C12, Set::Scalar C44, 
                                             Eigen::Matrix3d R = Eigen::Matrix3d::Identity())
     {
@@ -545,6 +557,8 @@ public:
                     }                
         return ret;
     }
+
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<3,Sym::MajorMinor> Cubic(Set::Scalar C11, Set::Scalar C12, Set::Scalar C44, 
                                             Set::Scalar phi1, Set::Scalar Phi, Set::Scalar phi2)
     {
@@ -554,6 +568,8 @@ public:
             Eigen::AngleAxisd(phi1, Eigen::Vector3d::UnitX());
         return Matrix4<3,Sym::MajorMinor>::Cubic(C11,C12,C44,R);
     }
+
+    AMREX_GPU_HOST_DEVICE
     static Matrix4<3,Sym::MajorMinor> Cubic(Set::Scalar C11, Set::Scalar C12, Set::Scalar C44, 
                                             Set::Quaternion q)
     {
@@ -624,14 +640,14 @@ public:
         return Transverse(C11,C12,C13,C33,C44,R);
     }
 
-    friend bool operator == (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b);
-    friend Matrix4<3,Sym::MajorMinor> operator + (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b);
-    friend Matrix4<3,Sym::MajorMinor> operator - (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b);
-    friend Matrix4<3,Sym::MajorMinor> operator * (Matrix4<3,Sym::MajorMinor> a, Set::Scalar b);
-    friend Matrix4<3,Sym::MajorMinor> operator * (Set::Scalar b, Matrix4<3,Sym::MajorMinor> a);
-    friend Matrix4<3,Sym::MajorMinor> operator / (Matrix4<3,Sym::MajorMinor> a, Set::Scalar b);
-    friend Set::Vector operator * (Matrix4<3,Sym::MajorMinor> a, Set::Matrix3 b);
-    friend Eigen::Matrix<amrex::Real,3,3> operator * (Matrix4<3,Sym::MajorMinor> a, Eigen::Matrix<amrex::Real,3,3> b);
+    friend AMREX_GPU_HOST_DEVICE bool operator == (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<3,Sym::MajorMinor> operator + (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<3,Sym::MajorMinor> operator - (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<3,Sym::MajorMinor> operator * (Matrix4<3,Sym::MajorMinor> a, Set::Scalar b);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<3,Sym::MajorMinor> operator * (Set::Scalar b, Matrix4<3,Sym::MajorMinor> a);
+    friend AMREX_GPU_HOST_DEVICE Matrix4<3,Sym::MajorMinor> operator / (Matrix4<3,Sym::MajorMinor> a, Set::Scalar b);
+    friend AMREX_GPU_HOST_DEVICE Set::Vector operator * (Matrix4<3,Sym::MajorMinor> a, Set::Matrix3 b);
+    friend AMREX_GPU_HOST_DEVICE Eigen::Matrix<amrex::Real,3,3> operator * (Matrix4<3,Sym::MajorMinor> a, Eigen::Matrix<amrex::Real,3,3> b);
 };
 AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE 
 bool operator == (Matrix4<3,Sym::MajorMinor> a, Matrix4<3,Sym::MajorMinor> b)
