diff --git a/src/Model/Gas/EOS/CPG.H b/src/Model/Gas/EOS/CPG.H
index e4866f1d5..50c0d5660 100644
--- a/src/Model/Gas/EOS/CPG.H
+++ b/src/Model/Gas/EOS/CPG.H
@@ -7,38 +7,51 @@
 
 namespace Model {
 namespace Gas {
-class Gas; // Forward declare Gas
 
 namespace EOS {
 
-class CPG : public EOS {
+class CPG
+{
 public:
     static constexpr const char* name = "cpg";
-    virtual const char* model_name() const override { return name; }
 
-private:
-    Gas const * const gas;
+public:
+    static void Parse(CPG & /*value*/, IO::ParmParse & /*pp*/) {}
 
 public:
-    CPG() = delete;
-    CPG(Gas *a_gas, IO::ParmParse& pp, std::string name) : gas(a_gas)
+    template<typename GAS>
+    AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
+    double ComputeT(const GAS &gas, double density, double momentumx, double momentumy, double E, double Tguess,
+                    Set::Patch<const Set::Scalar>& X, int i, int j, int k, double /*rtol=1e-12*/) const
     {
-        pp.queryclass(name, *this);
+        double P = (E - 0.5*(momentumx*momentumx + momentumy*momentumy)/density) * (gas.gamma(Tguess, X, i, j, k) - 1.0);
+        double T = P / density / gas.R(X, i, j, k);
+        return T;
     }
 
-    ~CPG() override = default;
-
-    static void Parse(CPG & /*value*/, IO::ParmParse & /*pp*/) {}
-
-public:
-    double ComputeT(double density, double momentumx, double momentumy, double E, double Tguess,
-                    Set::Patch<const Set::Scalar>& X, int i, int j, int k, double /*rtol=1e-12*/) const override;
+    AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
     double ComputeT(double pressure, double density,
-                    Set::Patch<const Set::Scalar>& X, int i, int j, int k) const override;
+                    double R) const
+    {
+        return pressure / density / R;
+    }
+
+    AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
     double ComputeP(double density, double T,
-                    Set::Patch<const Set::Scalar>& X, int i, int j, int k) const override;
+                    double R) const
+    {
+        return density * R * T;
+    }
+
+    AMREX_FORCE_INLINE AMREX_GPU_HOST_DEVICE
     double ComputeE(double density, double momentumx, double momentumy, double T,
-                    Set::Patch<const Set::Scalar>& X, int i, int j, int k) const override;
+                    double R, double gamma) const
+    {
+        double P = density * R * T;
+        double rhoE = P / (gamma - 1.0);
+        double E = rhoE + 0.5*(momentumx*momentumx + momentumy*momentumy)/density;
+        return E;
+    }
 
 }; // class CPG
 
@@ -47,3 +60,4 @@ public:
 } // namespace Model
 
 #endif
+
