diff --git a/src/Model/Gas/EOS/EOS.H b/src/Model/Gas/EOS/EOS.H
index f478f26b1..e177d5189 100644
--- a/src/Model/Gas/EOS/EOS.H
+++ b/src/Model/Gas/EOS/EOS.H
@@ -6,37 +6,101 @@
 #include <cmath>
 
 #include "Set/Set.H"
+#include "IO/ParmParse.H"
 
 namespace Model {
 namespace Gas {
 namespace EOS {
 
+template<typename... OBJ>
 class EOS {
-protected:
+private:
+    friend IO::ParmParse;
+    std::tuple<OBJ...> obj;
+    static constexpr const char* names[] = { OBJ::name... };
+    int selected = -1;
 
 public:
-    EOS() {} ;
-    virtual ~EOS() = default;
-    virtual const char* model_name() const = 0;
+    // Temperature, K
+    template<typename GAS, int I = 0>
+    AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
+    double ComputeT(const GAS &gas, double density, double momentumx, double momentumy, double E, double Tguess,
+                    Set::Patch<const Set::Scalar>& X, int i, int j, int k, double rtol) const
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).ComputeT(gas,density,momentumx,momentumy,E,Tguess,X,i,j,k,rtol);
+            else
+                return ComputeT<GAS, I+1>(gas,density,momentumx,momentumy,E,Tguess,X,i,j,k,rtol);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
 
     // Temperature, K
-    virtual double ComputeT(
-        double density, double momentumx, double momentumy, double E, double Tguess,
-        Set::Patch<const Set::Scalar>& X, int i, int j, int k, double rtol) const = 0;
-    virtual double ComputeT(
-        double pressure, double density,
-        Set::Patch<const Set::Scalar>& X, int i, int j, int k) const = 0;
+    template<int I = 0>
+    AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
+    double ComputeT(double pressure, double density, 
+                    double R) const
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).ComputeT(pressure, density, R);
+            else
+                return ComputeT<I+1>(pressure, density, R);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
+
+
     // Pressure, Pa
-    virtual double ComputeP(
-        double density, double T,
-        Set::Patch<const Set::Scalar>& X, int i, int j, int k) const = 0;
+    template<int I = 0>
+    AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
+    double ComputeP(double density, double T, double R) const
     // Energy, J/m^3
-    virtual double ComputeE(
-        double density, double momentumx, double momentumy, double T,
-        Set::Patch<const Set::Scalar>& X, int i, int j, int k) const = 0;
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).ComputeP(density, T, R);
+            else
+                return ComputeP<I+1>(density, T, R);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
 
+    template<int I = 0>
+    AMREX_FORCE_INLINE
+    AMREX_GPU_HOST_DEVICE
+    double ComputeE(double density, double momentumx, double momentumy, double T, 
+                    double R, double gamma) const
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).ComputeE(density, momentumx, momentumy, T, R, gamma);
+            else
+                return ComputeE<I+1>(density, momentumx, momentumy, T, R, gamma);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
 }; // class EOS
-
 } // namespace EOS
 } // namespace Gas
 } // namespace Model
