diff --git a/src/IC/Laminate.H b/src/IC/Laminate.H
index 5c06ae916..dac9bc099 100644
--- a/src/IC/Laminate.H
+++ b/src/IC/Laminate.H
@@ -5,6 +5,7 @@
 #ifndef IC_LAMINATE_H_
 #define IC_LAMINATE_H_
 
+#include "AMReX_Loop.H"
 #include "IC/IC.H"
 #include "Util/Util.H"
 #include "IO/ParmParse.H"
@@ -26,6 +27,8 @@ public:
 
     void Add(const int& lev, Set::Field<Set::Scalar>& a_field, Set::Scalar)
     {
+        ALAMO_GPU_UNSUPPORTED_HOST_LOOP("IC::Laminate");
+
         for (amrex::MFIter mfi(*a_field[lev], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi)
         {
             amrex::Box bx = mfi.tilebox();
@@ -33,7 +36,7 @@ public:
             amrex::IndexType type = a_field[lev]->ixType();
 
             amrex::Array4<Set::Scalar> const& field = a_field[lev]->array(mfi);
-            amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+            amrex::LoopConcurrentOnCpu(bx, [=] (int i, int j, int k) {
 
                 Set::Vector x = Set::Position(i, j, k, geom[lev], type);
 
