diff --git a/src/IC/Trig.H b/src/IC/Trig.H
index 2f38a8aa9..9c966cbbe 100644
--- a/src/IC/Trig.H
+++ b/src/IC/Trig.H
@@ -9,6 +9,7 @@
 
 #include <complex>
 
+#include "AMReX_Loop.H"
 #include "IO/ParmParse.H"
 #include "AMReX_Vector.H"
 #include "IC/IC.H"
@@ -55,6 +56,8 @@ public:
 
     virtual void Add(const int &lev, Set::Field<Set::Scalar> &a_field, Set::Scalar) override
     {
+        ALAMO_GPU_UNSUPPORTED_HOST_LOOP("IC::Trig scalar");
+
         bool cellcentered = (a_field[0]->boxArray().ixType() == amrex::IndexType(amrex::IntVect::TheCellVector()));
 
         const Set::Scalar AMREX_D_DECL(L1 = IC<Set::Scalar>::geom[lev].ProbHi()[0] - IC<Set::Scalar>::geom[lev].ProbLo()[0],
@@ -69,7 +72,7 @@ public:
             else bx = mfi.grownnodaltilebox();
 
             amrex::Array4<Set::Scalar> const &field = a_field[lev]->array(mfi);
-            amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k)
+            amrex::LoopConcurrentOnCpu(bx, [=] (int i, int j, int k)
             {
                 Set::Scalar AMREX_D_DECL(x1,x2,x3);
                 if (cellcentered)
@@ -109,6 +112,8 @@ public:
     }
     virtual void Add(const int &lev, Set::Field<Set::Vector> &a_field, Set::Scalar) override
     {
+        ALAMO_GPU_UNSUPPORTED_HOST_LOOP("IC::Trig vector");
+
         bool cellcentered = (a_field[0]->boxArray().ixType() == amrex::IndexType(amrex::IntVect::TheCellVector()));
 
         const Set::Scalar AMREX_D_DECL(L1 = IC<Set::Vector>::geom[lev].ProbHi()[0] - IC<Set::Vector>::geom[lev].ProbLo()[0],
@@ -123,7 +128,7 @@ public:
             else bx = mfi.grownnodaltilebox();
 
             amrex::Array4<Set::Vector> const &field = a_field[lev]->array(mfi);
-            amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k)
+            amrex::LoopConcurrentOnCpu(bx, [=] (int i, int j, int k)
             {
                 Set::Scalar AMREX_D_DECL(x1,x2,x3);
                 if (cellcentered)
