diff --git a/src/BC/Operator/Elastic/TensionTest.H b/src/BC/Operator/Elastic/TensionTest.H
index 0f66e6745..4e37d1643 100644
--- a/src/BC/Operator/Elastic/TensionTest.H
+++ b/src/BC/Operator/Elastic/TensionTest.H
@@ -39,8 +39,13 @@ public:
     ~TensionTest() {};
 
     using Constant::Init;
+#ifdef ALAMO_GPU
+    using Elastic::operator();
+    using Elastic::set;
+#else
     using Constant::operator();
     using Constant::set;
+#endif
 
     enum LoadingMode {TRAC,DISP};
 
