diff --git a/src/Model/Gas/Thermo/Thermo.H b/src/Model/Gas/Thermo/Thermo.H
index f0840f31b..bcb013019 100644
--- a/src/Model/Gas/Thermo/Thermo.H
+++ b/src/Model/Gas/Thermo/Thermo.H
@@ -5,26 +5,76 @@
 #include <memory>
 #include "Set/Set.H"
 #include <cmath>
+#include "IO/ParmParse.H"
+#include "Set/Set.H"
 
 namespace Model {
 namespace Gas {
 namespace Thermo {
 
+template<typename... OBJ>
 class Thermo {
-protected:
+private:
+    friend IO::ParmParse;
+    std::tuple<OBJ...> obj;
+    static constexpr const char* names[] = { OBJ::name... };
+    int selected = -1;
+
 
 public:
-    Thermo() {} ;
-    virtual ~Thermo() = default;
-    virtual const char* model_name() const = 0;
+    template<int I = 0>
+    AMREX_FORCE_INLINE
+    double cp_mol(double T, Set::Patch<const Set::Scalar>& X, int i, int j, int k) const
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).cp_mol(T,X,i,j,k);
+            else
+                return cp_mol<I+1>(T,X,i,j,k);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
+
+
+    template<int I = 0>
+    AMREX_FORCE_INLINE
+    double enthalpy_mol(double T, Set::Patch<const Set::Scalar>& X, int i, int j, int k) const
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).enthalpy_mol(T,X,i,j,k);
+            else
+                return enthalpy_mol<I+1>(T,X,i,j,k);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
 
-    // Specific heat (constant pressure), J/(kmol-K)
-    virtual double cp_mol(double T, Set::Patch<const Set::Scalar>& X, int i, int j, int k) const = 0;
-    // Specific enthalpy, J/kmol)
-    virtual double enthalpy_mol(double T, Set::Patch<const Set::Scalar>& X, int i, int j, int k) const = 0;
-    // specific entropy, J/(kmol-K)
-    virtual double entropy_mol(double T, Set::Patch<const Set::Scalar>& X, int i, int j, int k) const = 0;
+    template<int I = 0>
+    AMREX_FORCE_INLINE
+    double entropy_mol(double T, Set::Patch<const Set::Scalar>& X, int i, int j, int k) const
+    {
+        if constexpr (I < sizeof...(OBJ))
+        {
+            if (selected == I)
+                return std::get<I>(obj).entropy_mol(T,X,i,j,k);
+            else
+                return entropy_mol<I+1>(T,X,i,j,k);
+        }
+        else {
+            Util::Abort(INFO, selected);
+            return NAN;
+        }
+    }
 
+    ~Thermo() = default;
 }; // class Thermo
 
 } // namespace Thermo
