diff --git a/src/BC/BC.H b/src/BC/BC.H
index 32a8858a0..ccc288074 100644
--- a/src/BC/BC.H
+++ b/src/BC/BC.H
@@ -135,11 +135,11 @@ protected:
 namespace BCUtil
 {
 int ReadString(std::string bcstring);
-bool IsPeriodic(int bctype);
-bool IsNeumann(int bctype);
-bool IsReflectEven(int bctype);
-bool IsReflectOdd(int bctype);
-bool IsDirichlet(int bctype);
+AMREX_GPU_HOST_DEVICE bool IsPeriodic(int bctype);
+AMREX_GPU_HOST_DEVICE bool IsNeumann(int bctype);
+AMREX_GPU_HOST_DEVICE bool IsReflectEven(int bctype);
+AMREX_GPU_HOST_DEVICE bool IsReflectOdd(int bctype);
+AMREX_GPU_HOST_DEVICE bool IsDirichlet(int bctype);
 }
 
 }
