diff --git a/src/Util/BMP.H b/src/Util/BMP.H
index 9bf1ed07e..c79ee7a28 100644
--- a/src/Util/BMP.H
+++ b/src/Util/BMP.H
@@ -1,6 +1,7 @@
 #ifndef UTIL_BMP_H
 #define UTIL_BMP_H
 
+#include <limits>
 #include <string>
 #include <iostream>
 #include <vector>
@@ -26,7 +27,8 @@ public:
             nx = *(int*)&info[18];
             ny = *(int*)&info[22];
             
-            data.resize(nx*ny);
+            std::vector<std::array<int,3>> data_host;
+            data_host.resize(nx*ny);
             
             int row_padded = (nx*3 + 3) & (~3);
             unsigned char* chardata = new unsigned char[row_padded];
@@ -36,13 +38,29 @@ public:
                 if (retcode != row_padded) Util::Abort(INFO,"Error reading BMP file: ",row_padded," bytes expected but only ", retcode, " read.");
                 for(int i = 0; i < nx*3; i += 3)
                 {
-                    (*this)(i/3,j)[0] = (int)chardata[i+2]; // R
-                    (*this)(i/3,j)[1] = (int)chardata[i+1]; // G
-                    (*this)(i/3,j)[2] = (int)chardata[i];   // B
+                    data_host[i/3 + nx*j][0] = (int)chardata[i+2]; // R
+                    data_host[i/3 + nx*j][1] = (int)chardata[i+1]; // G
+                    data_host[i/3 + nx*j][2] = (int)chardata[i];   // B
+
+                    m_min[0] = std::min(m_min[0], data_host[i/3 + nx*j][0]);
+                    m_min[1] = std::min(m_min[1], data_host[i/3 + nx*j][1]);
+                    m_min[2] = std::min(m_min[2], data_host[i/3 + nx*j][2]);
+
+                    m_max[0] = std::max(m_max[0], data_host[i/3 + nx*j][0]);
+                    m_max[1] = std::max(m_max[1], data_host[i/3 + nx*j][1]);
+                    m_max[2] = std::max(m_max[2], data_host[i/3 + nx*j][2]);
                 }
             }
             fclose(f);
             delete[] chardata;
+
+            data.resize(data_host.size());
+            amrex::Gpu::copyAsync(  amrex::Gpu::hostToDevice,
+                                    data_host.begin(),
+                                    data_host.end(),
+                                    data.begin());
+            amrex::Gpu::streamSynchronize();
+            data_ptr = data.data();
         }
         else
         {
@@ -51,39 +69,39 @@ public:
     }
 
     AMREX_FORCE_INLINE
-    std::array<int,3> & operator () (int i,int j)
+    AMREX_GPU_DEVICE
+    const std::array<int,3> operator () (int i,int j) const
     {
         Util::Assert(INFO,TEST(i < nx)," i = ",i," nx = ", nx);
         Util::Assert(INFO,TEST(j < ny)," j = ",j," ny = ", ny);
-        return data[nx*j + i];
+        return data_ptr[nx*j + i];
     }
+
+    AMREX_GPU_HOST
     std::array<int,3> min()
     {
-        std::array<int,3> _min = {std::numeric_limits<int>::max(), std::numeric_limits<int>::max(), std::numeric_limits<int>::max()};
-        for (unsigned int i = 0; i < data.size(); i++)
-        {
-            if (_min[0] > data[i][0]) _min[0] = data[i][0];
-            if (_min[1] > data[i][1]) _min[1] = data[i][1];
-            if (_min[2] > data[i][2]) _min[2] = data[i][2];
-        }
-        return _min;
+        return m_min;
     }
+    AMREX_GPU_HOST
     std::array<int,3> max()
     {
-        std::array<int,3> _max = {0, 0, 0};
-        for (unsigned int i = 0; i < data.size(); i++)
-        {
-            if (_max[0] < data[i][0]) _max[0] = data[i][0];
-            if (_max[1] < data[i][1]) _max[1] = data[i][1];
-            if (_max[2] < data[i][2]) _max[2] = data[i][2];
-        }
-        return _max;
+        return m_max;
     }
 
 public:
     int nx, ny;
-private:
-    std::vector<std::array<int,3>> data;
+    std::array<int, 3> m_min = {
+        std::numeric_limits<int>::max(),
+        std::numeric_limits<int>::max(),
+        std::numeric_limits<int>::max()
+    };
+    std::array<int, 3> m_max = {
+        std::numeric_limits<int>::min(),
+        std::numeric_limits<int>::min(),
+        std::numeric_limits<int>::min()};
+
+    amrex::Gpu::DeviceVector<std::array<int,3>> data;
+    std::array<int,3> *data_ptr = nullptr;
 };
 }
 
