diff --git a/src/Util/PNG.H b/src/Util/PNG.H
index a97dc808b..ab40c2929 100644
--- a/src/Util/PNG.H
+++ b/src/Util/PNG.H
@@ -155,7 +155,8 @@ public:
     }
 
     [[nodiscard]]
-    std::array<Set::Scalar, 4> operator()(Set::Vector &x)
+    AMREX_GPU_HOST_DEVICE
+    std::array<Set::Scalar, 4> operator()(Set::Vector &x) const
     {
 #ifndef ALAMO_NOPNG
         Util::Assert(INFO, TEST(row_pointers), "Running IC without initialization...");
@@ -168,7 +169,7 @@ public:
         Set::Scalar img_dy = 1.0;
         
         //Set::Scalar x, y;
-        Set::Vector ximg;
+        Set::Vector ximg = Set::Vector::Zero();
 
         if (fit == Fit::Stretch)
         {
@@ -199,8 +200,10 @@ public:
 
 
 
-        ximg(0) = std::min(ximg(0), 1.0); ximg(1) = std::min(ximg(1), 1.0);
-        ximg(0) = std::max(ximg(0), 0.0); ximg(1) = std::max(ximg(1), 0.0);
+        ximg(0) = std::min(ximg(0), 1.0);
+        ximg(1) = std::min(ximg(1), 1.0);
+        ximg(0) = std::max(ximg(0), 0.0);
+        ximg(1) = std::max(ximg(1), 0.0);
 
         ximg(0) *= img_width;
         ximg(1) *= img_height;
